### ISSUE TITLE
 local.ERROR: ErrorException: Undefined index: border_bottom_width

### ISSUE THREAD

**Issue by Kemallyson** [dompdf/dompdf#1604]
I have a problem using the jimmyjs/laravel-report-generator. It throws the following error

 local.ERROR: ErrorException: Undefined index: border_bottom_width in C:\xampp\htdocs\unitmaster\vendor\dompdf\dompdf\src\Css\Style.php:1363,

What could the issue be?

**Comment by bsweeney** [comment]
No idea. Do you have a sample of the HTML+CSS?

**Comment by Kemallyson** [comment]
I have no sample HTML/CSS, its auto generated, I have implemented the code as below


        ```
$query_builder = Member::select(['member_no', 'full_name', 'identification_no', 'telephone_no', 'mobile_no', 'email_address'])
            ->orderBy($order_by);

        $columns = [
            'Member Number' => 'member_no',
            'Full Name' => 'full_name',
            'ID/Passport #' => 'identification_no',
            'Cell #' => 'mobile_no',
            'Tel. #' => 'telephone_no',
            'Email Address'
        ];

        return PdfReport::of($title, $meta, $query_builder, $columns)
            ->setPaper('a4')
            ->download('member_report.pdf');
```
as is in the documentation. The error I get is as below

![jimyjs](https://user-images.githubusercontent.com/14993642/33283955-7fdffe26-d3be-11e7-80ab-15e5b0fc0531.PNG)

**Comment by bsweeney** [comment]
The call is coming through the margin collapse logic, is this with the just-posted 0.8.2 release?

I don't use laravel so I'm just guessing but you should be able to access the instance of Dompdf, in which case you can call `output_html` to see what's being rendered. Maybe if you modify that last bit of code to something like:

```
return PdfReport::of($title, $meta, $query_builder, $columns)
    ->make()
    ->getDomPDF()
    ->output_html();
```

`make` returns the Dompdf instance (laravel-dompdf in this instance), `getDomPDF` is the laravel-dompdf method to return Dompdf, `output_html` is the Dompdf method to return the imported HTML document.

**Comment by Kemallyson** [comment]
Yes, its with the recent release. 

Your method rendered the html perfectly. Thank you a lot. 

Let me work on how to download the pdf output.

**Comment by bsweeney** [comment]
If you want to post the HTML we can look at why Dompdf is tripping over the margins.

**Comment by bsweeney** [comment]
OK, I can reproduce. Looks to be introduced with 0.8.0. Quick fix to Dompdf would be fairly trivial (just check for the existence of the key) but I want to make sure we're handling it correctly. I'll add for the next release.

**Comment by Kemallyson** [comment]
That is okay. Will be waiting.

Thank you

**Comment by yowmen2208** [comment]
thank you @bsweeney! i am facing the same problem too.

@Kemallyson may i ask if you are already done with the downloading the pdf?

**Comment by Kemallyson** [comment]
@yowmen2208 I haven't figured a way out but I used @barryvdh  laravel package to generate pdf n it worked well. But on  jimmyjs/laravel-report-generator, it throws the above error.

**Comment by yowmen2208** [comment]
@Kemallyson I see, Thank you so much for your reply.

**Comment by yowmen2208** [comment]
@Kemallyson can you give me an example on how you used @barryvdh laravel package to generate pdf? sorry im kinda newbie in laravel.

**Comment by Kemallyson** [comment]
@yowmen2208 You need to have html views that will be converted to pdf. I have implemented it as below

```
 $transaction = DB::table('trans')
            ->where('id', $id)
            ->first();

//load the view and pass the data you want displayed in the view
        $pdf = \PDF::loadView('reports.receipts.purchase_of_units', compact('company', 'transaction'));
        return $pdf->stream($transaction->member_no . 'purchase_receipt.pdf');
```

The on your view, access the data as shown

`<table border="0" cellpadding="5px" cellspacing="1px" width="80%" align="center">

        <tr>
            <td align="left">Receipt No :{{ $transaction->id }}</td>

            <td align="right">Member No :{{$transaction->member_no}} </td>
        </tr>
        <tr>
            <td align="right"></td>
            <td align="left">Value Date: {{$transaction->trans_date}}</td>
        </tr>
    </table>`

**Comment by agvstin** [comment]
@yowmen2208 @Kemallyson 

Got it working!

It looks like an issue with the border styles of the `.head_content` declared by `jimmyjs/laravel-report-generator`:

```css
border-style: none none ridge none;
```
 
I managed to avoid the issue by adding a `border-width` rule, using custom CSS:

```php
PdfReport::of($title, $meta, $qb, $columns)
   ->setCss([
          '.head-content' => 'border-width: 1px',
   ])->download('report');
```

Apparently `dompdf` needs a `border-width` whenever a`border-style` is defined.

Perhaps a `1px` border width can be used as default when only `border-style` is present?

**Comment by Kemallyson** [comment]
Thanks @agvstin let me get working now

**Comment by yowmen2208** [comment]
Im getting this result.


![image](https://user-images.githubusercontent.com/30574973/33813087-da2f63be-de5c-11e7-931a-ee819b2d305a.png)


even I use ->strea() or ->download()

**Comment by aimalamiri** [comment]
I'm getting the same error 
Her is my code

return PdfReport::of($title, $meta, $queryBuilder, $columns)
            ->setCss([
                '.head-content' => 'border-width: 1px',
            ])
            ->setPaper('a6')
            ->setOrientation('landscape')
            ->withoutManipulation()
            ->download('HICDP - All Projects Report');

**Comment by bsweeney** [comment]
The cause of the issue was because we were directly accessing style property methods and not using the default accessor (`__get`) when retrieving style properties in some instances within Dompdf. The generic accessor sets a default value for the style properties when none is defined. This was fixed in 5cc48383e6f88cc3b285b1950f0650886fab14a4.

Though we've corrected the issue in our own code, the calls are still accessible. Anyone attempting to call these methods directly would bypass the default-setting behavior and still run across the issue. Ultimately, we should probably make the style property methods private to avoid this type of issue.

Still, I'm closing since your immediate issue has been resolved.
